drivers/i2c/i2c_nrfx_twi.c

```diff
@@ -153,11 +153,7 @@ static int init_twi(struct device *dev, const nrfx_twi_config_t *config)
 			    CONFIG_I2C_INIT_PRIORITY,			\
 			    &i2c_nrfx_twi_driver_api)
 
-#ifdef CONFIG_I2C_0_NRF_TWI
-I2C_NRFX_TWI_DEVICE(0);
-#endif
 
-#ifdef CONFIG_I2C_1_NRF_TWI
-I2C_NRFX_TWI_DEVICE(1);
-#endif
+#define DRIVER_INST(i, _) _EVAL(CONFIG_I2C_##i##_NRF_TWI, (I2C_NRFX_TWI_DEVICE(i);), ())
+UTIL_EVAL(UTIL_REPEAT(TWI_COUNT, DRIVER_INST, /*not used */))
```

Comment:
I like the approach, however we should make it a bit easier to use.
Could we have single macro which generates the instances? Something like this:
```
DRIVER_INST(TWI_COUNT, CONFIG_I2C_##i##_NRF_TWI, (I2C_NRFX_TWI_DEVICE(i);))
```

Reply:
I'm pretty sure that it can be simplified further but let see others opinion about the proposal before moving it forward. One thing to note is that *TWI_COUNT* is actually not needed. We can put arbitrary max value there since what matter is presence of given config flag with ID.